libgstc/javascript/libgstc.js

```diff
@@ -567,24 +539,65 @@ class GstdClient {
    * @param {String} signal
    * @param {String} timeout
    *
+   * @throws {GstdError} Error is triggered when Gstd IPC fails.
+   * @throws {GstcError} Error is triggered when GstClient fails.
+   *
    * @return {object} Response from Gstd.
    */
   async signal_timeout(pipe_name, element, signal, timeout) {
+
     var url = this.ip + ":" + this.port + "/pipelines/" + pipe_name +
-      "/elements/" + element + "/signals/" + signal + "timeout?name=" +
-      timeout;
-    var request = {
-      method: 'PUT',
-      body: {
-        name : pipe_name,
-        element : element,
-        signal : signal,
-        timeout : timeout
-      },
-    }
+      "/elements/" + element + "/signals/" + signal + "/timeout";
+    return this.update("/debug/threshold", timeout);
+  }
 
-    return GstdClient.send_cmd(url, request);
+  /**
+   * Send an end-of-stream event.
+   *
+   * @param {String} pipe_name.
+   *
+   * @throws {GstdError} Error is triggered when Gstd IPC fails.
+   * @throws {GstcError} Error is triggered when GstClient fails.
+   *
+   * @return {object} Response from Gstd.
+   */
+  async event_eos(pipe_name) {
+
+    return this.create("/pipelines/" + pipe_name + "/event", "eos");
+  }
+
+  /**
+   * Put the pipeline in flushing mode.
+   *
+   * @param {String} pipe_name.
+   *
+   * @throws {GstdError} Error is triggered when Gstd IPC fails.
+   * @throws {GstcError} Error is triggered when GstClient fails.
+   *
+   * @return {object} Response from Gstd.
+   */
+  async event_flush_start(pipe_name) {
+
+    var uri = "/pipelines/" + pipe_name + "/event";
+    return this.create(uri, "flush_start");
```

Comment:
Wont this fail? I think create must receive 3 arguments.

Reply:
It didn't fail because it was set to undefine. To make it clearer I made it explicitly undefined.